From 33c0f472768f3901cf2d60d19a30c81b138b418f Mon Sep 17 00:00:00 2001
From: Anton Joubert <1346825-ajoubertza@users.noreply.gitlab.com>
Date: Wed, 19 Jun 2024 16:54:16 +0200
Subject: [PATCH] Fix psutil DeprecationWarning

Since v6.0.0, `psutil.Process.connections` is deprecated.
The replacement is `net_connections`.  We use it if available,
but fallback to the old method if not.  That way we don't need
to restrict the version of psutil installed with PyTango.

The deprecation warning causes tests to fail since
we have configured pytest to convert warnings to errors.
---
 tango/test_context.py | 5 ++++-
 1 file changed, 4 insertions(+), 1 deletion(-)

diff --git a/tango/test_context.py b/tango/test_context.py
index f1078a05..7d6ae6f5 100644
--- a/tango/test_context.py
+++ b/tango/test_context.py
@@ -135,7 +135,10 @@ def get_server_port_via_pid(pid, host, retries=400, delay=0.03):
 
 def _get_listening_tcp_ports(pid):
     p = psutil.Process(pid)
-    conns = p.connections(kind="tcp")
+    if hasattr(p, "net_connections"):
+        conns = p.net_connections(kind="tcp")
+    else:
+        conns = p.connections(kind="tcp")  # deprecated in psutil v6.0.0
     return list(set([c.laddr[1] for c in conns]))
 
 
-- 
2.43.0

